server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -151,18 +145,7 @@ static boolean executeInternal(SearchContext searchContext) throws QueryPhaseExe
 
                 } else {
                     final ScoreDoc after = scrollContext.lastEmittedDoc;
-                    if (returnsDocsInOrder(query, searchContext.sort())) {
```

Comment:
Is the idea here that the sort optimization does this for us instead?

Reply:
Yes and the tests in QueryPhaseTests verifies that we do skip documents.